src/routes/safe/components/Apps/hooks/useLegalConsent.ts

```diff
@@ -3,15 +3,17 @@ import { loadFromStorage, saveToStorage } from 'src/utils/storage'
 
 const APPS_LEGAL_CONSENT_RECEIVED = 'APPS_LEGAL_CONSENT_RECEIVED'
 
-const useLegalConsent = (): { consentReceived: boolean; onConsentReceipt: () => void } => {
-  const [consentReceived, setConsentReceived] = useState<boolean>(false)
+const useLegalConsent = (): { consentReceived: boolean | undefined; onConsentReceipt: () => void } => {
+  const [consentReceived, setConsentReceived] = useState<boolean | undefined>()
 
   useEffect(() => {
     const checkLegalDisclaimer = async () => {
       const storedConsentReceived = await loadFromStorage(APPS_LEGAL_CONSENT_RECEIVED)
 
       if (storedConsentReceived) {
         setConsentReceived(true)
+      } else {
+        setConsentReceived(false)
```

Comment:
how does it compare to the previous implementation in functionality? (I don't understand the change)

Reply:
The functionality remains the same, this change was to avoid a flicker showing briefly the disclaimer.

Reply:
Ah, got it. I would move it to a separate flag (loaded),, though, so it's  more obvious